### PR TITLE
chore: upgrade to ubuntu 22.04 LTS + python3.10

### PR DESCRIPTION
**Pull request by ebezzi**
Required to fix our builds.

### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

### CODE REVIEW

File: backend/corpora/dataset_processing/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+numba==0.53.1
```
Comment (maniarathi) [inline backend/corpora/dataset_processing/requirements.txt:1]: Nit: can we please alphabetize `requirements.txt` docs? Makes it a lot easier to read.

File: Dockerfile [hunk 2]
```diff
@@ -12,11 +12,9 @@ RUN apt-get update && \
 
 # Don't re-run pip install unless either requirements.txt has changed.
 WORKDIR /single-cell-data-portal
-ADD requirements.txt /single-cell-data-portal/requirements.txt
-ADD backend/corpora/api_server/requirements.txt /single-cell-data-portal/requirements-api.txt
-RUN grep -v requirements.txt requirements.txt > reqs.txt \
-    && cat requirements-api.txt >> reqs.txt \
-    && python3 -m pip install -r reqs.txt
+ADD requirements.txt requirements.txt
```
Comment (atolopko-czi) [inline Dockerfile:15]: for consistency across Dockerfiles, perhaps this:
```suggestion
ADD requirements.txt requirements-base.txt
```

File: requirements.txt [hunk 3]
```diff
@@ -8,23 +8,21 @@ coverage>=2.0.15
...
 PyMySQL==0.9.3
 pytest
 requests>=2.22.0
 s3fs==0.4.2
 scanpy
-scipy==1.7.3
+#scipy==1.7.3
 tenacity
-tiledb==0.13.2  # Portal's tiledb version should always be the same or older than Explorer's
--r ./backend/corpora/api_server/requirements.txt
```
Comment (atolopko-czi) [inline requirements.txt:27]: love this removal

**Review by atolopko-czi — APPROVED** [review]
LGTM.   Happy to review again, since I understand more changes in the pipeline [pun intended].

File: requirements.txt (hunk 3 above)
Comment (ebezzi) [reply requirements.txt:27]: Yup, that seemed a bit hacky.

**Review by atolopko-czi — APPROVED** [review]
Thanks for pushing through on this one!

**Pull request merged**